lib/dolly/document.rb

```diff
@@ -128,7 +128,8 @@ def write_property name, value
     end
 
     def read_property name
-      self.properties[name].value
+      return self.properties[name].value unless doc[name.to_s]
```

Comment:
> should take a look at the property class and see if I hide the logic in there?

I tried that approach and didn't went well as the instance have not been initialize when the property object is created.

but yeah maybe you can hide the logic in property object where you set a doc through delegation into Property, but this will be set after initialization... of document, that is the hard part